41-vaccine-temp-monitor/README.md

```diff
@@ -0,0 +1,321 @@
+synchronization event (max wait: inbound frequency) for the changes to propagate
+to the host._
+
+Operation
+---------
+
+At this point, the host firmware should be running and you should start to see
+events show up on Notehub under your project's "Events" view.
+
+The primary Notefile associated with the application is `status.qo`.
+
+### Normal Operation
+
+#### Temperature Sampling
+
+By default, the host will sample time stamped temperature data from the LM75A to
+`status.qo` every 30 minutes. You can change this interval by setting the
+environment variable `sample_interval_s`. Regardless of the sampling interval,
+the host will publish the collected data to Notehub based on the amount of
+charge remaining on the Li-Ion battery (depicted in table below).
+
+| Charge Level | Outbound Freq (min) | Inbound Freq (min) |
+|:------------:|:-------------------:|:------------------:|
+|   ON POWER   |           5         |          5         |
+|     HIGH     |          60         |        120         |
+|    NORMAL    |         120         |        240         |
+|     LOW      |         240         |        480         |
+|     DEAD     |           0         |          0         |
+
+_**NOTE:** Reducing `sample_interval_s` will cause the application to sample the
+temperature and battery condition more frequently and will shorten the expected
+battery life of the device._
+
+Normal data:
+
+```json
+{
+    "temp":18.5,
+    "voltage":3.5898438
+}
+```
+
+#### Location sampling
+
+The device is configured to use motion based GPS tracking. When motion is
+detected, the device will report its location periodically; every 15 minutes.
+Regardless of whether or not the device moves, it will also report a heartbeat
+location every 12 hours.
+
+### Alarm Conditions
+
+#### Thermal Event
+
+By default, the firmware will _IMMEDIATELY_ send an alarm if the temperature
+exceeds `alarm_temperature`. An alarm condition will cause the LM75A will trip
+the `OS` pin, which alerts the Notecard and causes it to wake the host from
+sleep.
+
+Temp alert data:
+
+```json
+{
+    "alert":true,
+    "temp":30.5,
+    "voltage":3.5898438
+}
+```
+
+The `alert` field is added to the JSON object, when the temperature has exceeded
+the `alarm_temperature` environment variable value. When the `alert` key is
+received by Notehub, then an SMS message containing the temperature data and GPS
+coordinates of the device is forwared to Twilio. There a text message is
+generated and sent to the phone number specified in the `alarm_sms_number`
+environment variable.
+
+##### Thermal Event Operation
+
+During a thermal event, time is of the essense and the mode of operation changes
+accordingly. Instead of sampling every 30 minutes, the temperature and location
+will be sampled every every 3 minutes. While in this mode, the device will force
+a GPS fix or timeout after 95 seconds.
+
+#### Low Battery
+
+When `card.voltage` returns `"mode":"low"`, then the `low_batt` key is added to
```

Comment:
What is card.voltage? I think we should assume the reader doesn't know anything about Notecard requests.